sqlest/src/main/scala/sqlest/ast/syntax/UntypedReads.scala

```diff
@@ -0,0 +1,72 @@
+package sqlest.ast.syntax
+
+import org.joda.time.{ DateTime, LocalDate }
+import scala.util.Try
+import sqlest.ast._
+import sqlest.util.Iso8601
+
+trait UntypedReads[A] {
```

Comment:
I'm not convinced that both the type param and the type member are required. Aren't they always going to be the same thing? Would this work?

```scala
trait UntypeReads[A] {
  def reads(s: String): Option[A]
}
```